pkg/provider/azure.go

```diff
@@ -573,14 +573,18 @@ func (az *Cloud) InitializeCloudFromConfig(config *Config, fromSecret, callFromC
 		az.routeUpdater = newDelayedRouteUpdater(az, routeUpdateInterval)
 		go az.routeUpdater.run()
 
-		// wait for the success first time of syncing zones
-		err = az.syncRegionZonesMap()
-		if err != nil {
-			klog.Errorf("InitializeCloudFromConfig: failed to sync regional zones map for the first time: %s", err.Error())
-			return err
-		}
+		// Azure Stack does not support zone at the moment
+		// https://docs.microsoft.com/en-us/azure-stack/user/azure-stack-network-differences?view=azs-2102
+		if !az.isStackCloud() {
+			// wait for the success first time of syncing zones
+			err = az.syncRegionZonesMap()
+			if err != nil {
+				klog.Errorf("InitializeCloudFromConfig: failed to sync regional zones map for the first time: %s", err.Error())
+				return err
+			}
```

Comment:
Nit: This would be more idiomatic. When a function returns a single err variable it's common to inline the if statement. Also, use `%v` for errors, it has the same effect as `%s` and `err.Error()`
```suggestion
			if err := az.syncRegionZonesMap(); err != nil {
				klog.Errorf("InitializeCloudFromConfig: failed to sync regional zones map for the first time: %v", err)
				return err
			}
```

Reply:
This code from upstream, I just wrapped it with stackcloud check. Would prefer to leave it as is.